connmgr/seed.go

```diff
@@ -41,7 +41,7 @@ func SeedFromDNS(chainParams *chaincfg.Params, reqServices wire.ServiceFlag,
 			host = fmt.Sprintf("x%x.%s", uint64(reqServices), dnsseed.Host)
 		}
 
-		go func(host string) {
+		go func(host string, services wire.ServiceFlag) {
```

Comment:
I don't think this is right since the `Services` field is for alll the services that the address supports. https://github.com/utreexo/utreexod/blob/affee1685ed328d771dbf01aa1d0a2e13bab84d2/wire/netaddress.go#L69-L70

If we add `reqServices` to it, it's not entirely correct since it's not all the supported services that the address has. The services are set during the version negotiation in here: https://github.com/btcsuite/btcd/blob/80f5a0ffdf363cfff27d550f9e38aa262667a7f1/server.go#L477-L479

Since the services are set elsewhere, the change made here isn't really going to do much and I'd prefer to leave it as is.

Reply:
My goal here was to tell the daemon to connect with utreexo peers before any other. It's not like we won't set this up on handshake, but if the `dnsseed` told this peer has a given flag, it's almost certain that it have. Therefore, we can use this as a hint to open utreexo connections at startup, without having to try many peers at random. But since `openconn` don't use this, it makes no sense. 

Is it reasonable to change the connection logic to try utreexo peers before non-utreexo peers if `-utreexo` is set? I did this on Floresta, and it opens an `utreexo` connection right at startup.

Reply:
> Is it reasonable to change the connection logic to try utreexo peers before non-utreexo peers if `-utreexo` is set?

Yeah that's reasonable.

> My goal here was to tell the daemon to connect with utreexo peers before any other.

I'm not quite sure if that's what this code is achieving. Could you let me know where I should look to verify that passing in the services flag will make the node try a utreexo node first?

Reply:
> I'm adding this to tell which services are supported by each peer, so we can connect with utreexo peers right away. But it seems connmgr isn't using it at all?

Took a look at the code and you're right, connmgr isn't really differentiating between nodes based on their services. Seems like it's something we can address later on but would it be ok to leave it out of this PR? I prefer if PRs achieved one singular goal and this PR's goal is to add an utreexo dns seed, not to change the connmgr logic